### ISSUE TITLE
not supporting laravel 5.6.4

### ISSUE THREAD

**Issue by amanullah-1** [tymondesigns/jwt-auth#1486]
### Subject of the issue
I ma trying to install it on laravel 5.6.4 but not being able

### Your environment
| Q                 | A
| ----------------- | ---
| Bug?              | no 
| New Feature?      |yes
| Framework         | Laravel 
| Framework version | 5.6.4
| Package version   | 0.5.12
| PHP version       | 7.x.y

### Steps to reproduce
running the command: 
composer require tymon/jwt-auth

### Expected behaviour
to be installed

### Actual behaviour
showing error: 
Using version ^0.5.12 for tymon/jwt-auth
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Conclusion: don't install barryvdh/laravel-cors v0.8.6
    - Conclusion: don't install barryvdh/laravel-cors v0.8.5
    - Conclusion: don't install barryvdh/laravel-cors V0.8.4
    - Conclusion: don't install barryvdh/laravel-cors v0.8.3
    - Conclusion: remove laravel/framework v5.6.3
    - Installation request for barryvdh/laravel-cors ^0.8.2 -> satisfiable by barryvdh/laravel-cors[V0.8.4, v0.8.2, v0.8.3, v0.8.5, v0.8.6].
    - Conclusion: don't install laravel/framework v5.6.3
    - barryvdh/laravel-cors v0.8.2 requires illuminate/support 5.1.x|5.2.x|5.3.x -> satisfiable by illuminate/support[5.1.x-dev, 5.2.x-dev, 5.3.x-dev, v5.1.1, v5.1.13, v5.1.16, v5.1.2, v5.1.20, v5.1.22, v5.1.25, v5.1.28, v5.1.30, v5.1.31, v5.1.41, v5.1.6, v5.1.8, v5.2.0, v5.2.19, v5.2.21, v5.2.24, v5.2.25, v5.2.26, v5.2.27, v5.2.28, v5.2.31, v5.2.32, v5.2.37, v5.2.43, v5.2.45, v5.2.6, v5.2.7, v5.3.0, v5.3.16, v5.3.23, v5.3.4].
    - don't install illuminate/support 5.1.x-dev|don't install laravel/framework v5.6.3
    - don't install illuminate/support 5.2.x-dev|don't install laravel/framework v5.6.3
    - don't install illuminate/support 5.3.x-dev|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.1.1|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.1.13|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.1.16|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.1.2|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.1.20|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.1.22|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.1.25|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.1.28|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.1.30|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.1.31|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.1.41|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.1.6|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.1.8|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.2.0|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.2.19|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.2.21|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.2.24|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.2.25|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.2.26|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.2.27|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.2.28|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.2.31|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.2.32|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.2.37|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.2.43|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.2.45|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.2.6|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.2.7|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.3.0|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.3.16|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.3.23|don't install laravel/framework v5.6.3
    - don't install illuminate/support v5.3.4|don't install laravel/framework v5.6.3
    - Installation request for laravel/framework (locked at v5.6.3, required as 5.6.*) -> satisfiable by laravel/framework[v5.6.3].

Installation failed, reverting ./composer.json to its original content.

**Update:**

Firstly I was trying composer update command adding some specific version of dependencies in composer.json file. Like "tymon/jwt-auth": "0.5.*" and "barryvdh/laravel-cors": "^0.8.2".

After that I was trying composer require tymon/jwt-auth not removing the lines with specific version of tymon/jwt-auth and barryvdh/laravel-cors. After removing the lines from composer.json file with specific versions of tymon/jwt-auth the

composer require tymon/jwt-auth  
command is working fine.

**Comment by masterawss** [comment]
https://github.com/tymondesigns/jwt-auth/issues/1038

I used the command "tymon / jwt-auth": "1.0.0-rc.2" and it worked fine for the @cieniurobot solution

**Comment by josh9060** [comment]
JWT 0.5.* is only supported on Laravel versions 5.5 or less. Check out the 0.5.* docs. If you want to use JWT with 5.6.*, use JWT 1.0.*.

**Comment by dellamina** [comment]
I was facing the same issues, reinstalling directly with composer and not laravel installer fixed it for me.

**Comment by rodrigobendia** [comment]
@amanullah-1 In your case, the issue seems to be with the barryvdh/laravel-cors package. The installed version (v0.8.2) is forbidding composer to install Laravel 5.6. Try running _composer update_ before requiring this package.

**Comment by musman92** [comment]
yes, i tried to, but stuck, i am unable to use web/api guards. donot know what to do now, 

also get this error 
Method Illuminate\Auth\SessionGuard::factory does not exist.

**Comment by GizzmoShifu** [comment]
Setting the dependencies to the following worked for me:

`"tymon/jwt-auth": "1.0.0-rc.2",`
`"barryvdh/laravel-cors": "^0.11.0"`

**Comment by adriiiprieto** [comment]
@GizzmoAsus Great! That worked for me.

**Comment by drmmr763** [comment]
Just a +1 from the suggestion from @GizzmoAsus. I recommend putting this into documentation somewhere until 1.0 stable is released.

**Comment by iamirfanfaiz** [comment]
The following approach worked for me.

On **composer.json** file enter the latest version of **jwt-auth**:
`"tymon/jwt-auth": "1.0.0-rc.4.1"`

and run:
`composer update`
